target/arm/cpu.h

```diff
@@ -2768,6 +2937,24 @@ static inline bool cpreg_field_is_64bit(const ARMCPRegInfo *ri)
     return (ri->state == ARM_CP_STATE_AA64) || (ri->type & ARM_CP_64BIT);
 }
 
+static inline bool cpreg_field_is_cap(const ARMCPRegInfo *ri)
+{
+#ifdef TARGET_CHERI
+    return !!(ri->type & ARM_CP_CAP);
```

Comment:
They are semantically different, as ARM_CP_CAP_ONLY is not a single bit. I will make this the same as below anyway as for the single bit it is the same, and it may change in the future.